src/h3/mod.rs

```diff
@@ -826,6 +826,12 @@ impl Connection {
     /// [`send_body()`]: struct.Connection.html#method.send_body
     /// [`close()`]: ../struct.Connection.html#method.close
     pub fn poll(&mut self, conn: &mut super::Connection) -> Result<(u64, Event)> {
+        // When the connection is closing (e.g. due to a protocol error) it might
+        // be in a broken state, so return early.
+        if conn.is_closing() {
+            return Err(Error::Done);
+        }
```

Comment:
Just wondering how this would relate to TODO GOAWAY behaviour? 

Imagine the scenario where a client sends 10 requests that are received and buffered up ready to be received via `poll()`, it issues a GOAWAY with MAX INT, sends all the responses, then sends CONNECTION_CLOSE(NO_ERROR). Would this early close cause all the pending response data to be "stuck" inside the library and blown away before the client could read it?

Reply:
That's regardless of this change I think. If `close()` is called we won't send `STREAM` frames anymore https://github.com/cloudflare/quiche/blob/530b207197b33631d264dedaf69ab3009e3f5529/src/lib.rs#L1842

Not sure how `GOAWAY` is supposed to work though. If the server cares about the data being received by the client it really needs to send `CONNECTION_CLOSE` only after the data has been sent (e.g. application could be blocked because of not enough flow control) and acked.

In HTTP/2 I don't think we close the TCP socket immediately after sending `GOAWAY` in most cases, so I guess HTTP/3 should work like that as well.

Reply:
So in this case the data would be "received" by the client because it was read out of the QUIC layer, but at some point a CONNECTION_CLOSE could come in disrupt the flushing of our H3 layer queuing. We can probably handle that by modulating the check to consider graceful close when we get around to implementing it. 

For now I think the PR makes things better.